// Copyright 2017 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";
option cc_enable_arenas = true;

package ray.rpc;

import "src/ray/protobuf/common.proto";

message GetObjectRequest {
  repeated bytes object_ids = 1;

  repeated unsigned long virt_address = 2;
  repeated int32 object_size = 3;
  repeated int32 meta_size = 4;
  repeated bytes owner_raylet_id = 5;
  repeated string owner_ip_address = 6;
  repeated int32 owner_port = 7;
  repeated bytes owner_worker_id = 8;
  repeated string rem_ip_address = 9;
}

// Reply for request
message GetObjectReply {
}

service ObjectManagerRdmaService {
  // Tell remote object manager rdma to get some objects
  rpc GetObject(GetObjectRequest) returns (GetObjectReply);
}
